src/tests/test_utils/common_test_utils/src/ov_tensor_utils.cpp

```diff
@@ -314,141 +314,105 @@ inline double less_or_equal(double a, double b) {
+        value <= std::numeric_limits<T>::min()) {
+        return false;
+    }
+    return true;
+}
+
+class Error {
+protected:
+    struct IncorrectValue {
+        size_t coordinate;
+        double actual_value, expected_value, abs_threshold, rel_threshold;
+
+        IncorrectValue(double in_actual_value,
+                       double in_expected_value,
+                       double in_abs_threshold,
+                       double in_rel_threshold,
+                       size_t in_coordinate)
+            : actual_value(in_actual_value),
+              expected_value(in_expected_value),
+              abs_threshold(in_abs_threshold),
+              rel_threshold(in_rel_threshold),
+              coordinate(in_coordinate) {}
+    };
 
-    Error(double _threshold) : threshold(_threshold) {}
+    std::vector<IncorrectValue> incorrect_values;
+    double abs_threshold, rel_threshold;
 
-    void update(double val, size_t i) {
-        if (less(max, val)) {
-            max = val;
-            max_coordinate = i;
+public:
+    Error(const double in_abs_threshold, const double in_rel_threshold)
+        : abs_threshold(in_abs_threshold),
+          rel_threshold(in_rel_threshold) {}
+
+    void update(double actual, double expected, size_t coordinate) {
+        const auto diff = std::fabs(expected - actual);
+
+        const auto calculated_abs_threshold = abs_threshold * expected;
+        const auto calculated_rel_threshold = calculated_abs_threshold ? diff / calculated_abs_threshold : 1.;
+        if (less_or_equal(diff, calculated_abs_threshold) && less_or_equal(calculated_rel_threshold, rel_threshold)) {
+            return;
         }
-        mean += val;
-        count += less(threshold, val);
+        incorrect_values.emplace_back(
+            IncorrectValue(actual, expected, calculated_abs_threshold, calculated_rel_threshold, coordinate));
     }
-};
 
-inline double calculate_median(std::vector<double>& abs_values) {
-    double abs_median = 0.;
-    auto expected_shape = abs_values.size();
-    if (expected_shape % 2) {
-        std::nth_element(abs_values.begin(), abs_values.begin() + expected_shape / 2, abs_values.end());
-        abs_median = abs_values[expected_shape / 2];
-    } else {
-        std::nth_element(abs_values.begin(), abs_values.begin() + expected_shape / 2, abs_values.end());
-        std::nth_element(abs_values.begin(), abs_values.begin() + (expected_shape - 1) / 2, abs_values.end());
-        abs_median = (abs_values[(expected_shape - 1) / 2] + abs_values[expected_shape / 2]) / 2.0;
+    void get_results() {
+        if (!incorrect_values.empty()) {
+            std::string msg = "[ COMPARATION ] COMPARATION IS FAILED! incorrect elem counter: ";
+            msg += std::to_string(incorrect_values.size());
+            msg += ". Please print `incorrect_values` to get detailed information!";
+            throw std::runtime_error(msg);
+        }
```

Comment:
As kind of optimization, let's collect full `incorrect_values`only in Debug. In Release it's only time/memory consuming and affects execution

It may be implemented like:
```
for (size_t i = 0; i < shape_size_cnt; ++i) {
    ...
    auto status = error.update(expected_value, actual_value, i);
#if !NDEBUG
    if (!status)
        break;
#
}
```

Reply:
done